src/security/core/tests/CMakeLists.txt

```diff
@@ -11,14 +11,44 @@
 #
 include (GenerateExportHeader)
 include (CUnit)
+include (Generate)
 
 idlc_generate(TARGET SecurityCoreTests FILES SecurityCoreTests.idl WARNINGS no-implicit-extensibility)
 
 function(add_wrapper libname linklibs)
   set(srcs_wrapper
     "${CMAKE_CURRENT_LIST_DIR}/common/${libname}_wrapper.c"
     "${CMAKE_CURRENT_LIST_DIR}/common/plugin_wrapper_msg_q.c")
-  add_library("dds_security_${libname}_wrapper" SHARED "")
+  add_library("dds_security_${libname}_wrapper")
+  if(NOT BUILD_SHARED_LIBS)
+    set_property(GLOBAL APPEND PROPERTY cdds_plugin_list dds_security_${libname}_wrapper)
+    set_property(GLOBAL PROPERTY dds_security_${libname}_wrapper_symbols
+      init_test_${libname}_wrapped finalize_test_${libname}_wrapped
+      init_test_${libname}_missing_func finalize_test_${libname}_missing_func
+      init_test_${libname}_all_ok finalize_test_${libname}_all_ok)
+    if(libname STREQUAL authentication)
+      set_property(GLOBAL APPEND PROPERTY dds_security_${libname}_wrapper_symbols
+        init_test_${libname}_init_error finalize_test_${libname}_init_error)
+    elseif(libname STREQUAL access_control)
+      set_property(GLOBAL APPEND PROPERTY dds_security_${libname}_wrapper_symbols
+        init_test_${libname}_local_participant_not_allowed
+        init_test_${libname}_local_topic_not_allowed
+        init_test_${libname}_local_writer_not_allowed
+        init_test_${libname}_local_reader_not_allowed
+        init_test_${libname}_local_permissions_not_allowed
+        init_test_${libname}_remote_participant_not_allowed
+        init_test_${libname}_remote_topic_not_allowed
+        init_test_${libname}_remote_writer_not_allowed
+        init_test_${libname}_remote_reader_not_allowed
+        init_test_${libname}_remote_reader_relay_only
+        init_test_${libname}_remote_permissions_not_allowed
+        finalize_test_${libname}_not_allowed)
```

Comment:
Updating these symbol lists can easily be missed when adding functions, but I guess that leads to an error in the static build on CI?

Reply:
Yes, leaving them out means that `dlsym` will fail to find the symbols and that causes test failures. It would be nicer to automatically generate this list, but then I need to figure out a way to build the library first, then extract the symbols, add them to the plug-in properties, then continue building the library. I'd have done it if it were normal `make` ... Maybe one day I can find the courage to try it with CMake!